test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs

```diff
@@ -141,10 +143,16 @@ public void DataCollectorAttachmentProcessor(RunnerInfo runnerInfo)
                          .Add(new XElement("Configuration", new XElement("MergeFile", "MergedFile.txt")));
             runSettingsXml.Save(runSettings);
 
-            this.InvokeVsTest(arguments);
+            var env = new Dictionary<string, string>
+            {
+                ["SampleDataCollectorTempPath"] = workspace.Path,
+            };
+
+            this.InvokeVsTest(arguments, env);
+
             this.ValidateSummaryStatus(2, 2, 2);
 
-            string mergedFile = Directory.GetFiles(resultsDir, "MergedFile.txt", SearchOption.AllDirectories).Single();
+            string mergedFile = Directory.GetFiles(workspace.Path, "MergedFile.txt", SearchOption.AllDirectories).Single();
```

Comment:
we can also think to add GetFiles to workspace...it's a common task search for files.

Reply:
Yup, I had that in my original proposal as well. So maybe as a second step?